source/code/plugins/out_oms_diag.rb

```diff
@@ -0,0 +1,112 @@
+module Fluent
+
+  class OutputOMSDiag < BufferedOutput
+
+    Plugin.register_output('out_oms_diag', self)
+
+    def initialize
+      super
+      require 'net/http'
+      require 'net/https'
+      require 'uri'
+      require_relative 'omslog'
+      require_relative 'oms_configuration'
+      require_relative 'oms_common'
+      require_relative 'oms_diag'
+    end
+
+    config_param :omsadmin_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/conf/omsadmin.conf'
+    config_param :cert_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.crt'
+    config_param :key_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.key'
+    config_param :proxy_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/proxy.conf'
+    # TODO: To make compress default to true once supported
```

Comment:
The link implies that we would require changes to the DSC module to update it. I was going for not changing something in the code for DSC for this plugin. It will be downloaded on an update to module version of Common plugins and can have the change without altering the omsagent.conf.